lib/resource-tagger.js

```diff
@@ -0,0 +1,35 @@
+const assert = require('assert');
+
+class ResourceTagger {
+
+  constructor({runaws, ec2, keyPrefix}) {
+    assert(typeof runaws === 'function');
+    this.runaws = runaws;
+
+    assert(typeof ec2 === 'object');
+    this.ec2 = ec2;    
+ 
+    assert(typeof keyPrefix === 'string');
+    assert(keyPrefix[keyPrefix.length - 1] === ':');
```

Comment:
Should group type-checking asserts at the top, then constructing logic underneath